src/words/ast/INodeMovesPredicate.java

```diff
@@ -10,7 +15,22 @@ public INodeMovesPredicate(Object... children) {
 
 	@Override
 	public ASTValue eval(WordsEnvironment environment) throws WordsRuntimeException {
-		// TODO
-		throw new AssertionError("Not yet implemented");
+		ASTValue className = children.get(0).eval(environment);
+		ASTValue objectAlias = children.get(1).eval(environment);
+		ASTValue moveDirection = children.get(2) == null ? null : children.get(2).eval(environment);
+		
+		HashSet<WordsObject> objectsOfClass = environment.getObjectsByClass(className.stringValue);
+
+		for (WordsObject object : objectsOfClass) {
```

Comment:
I have a concern here.  It looks like this implementation causes a Moves predicate to return true whenever *any* object of the class has moved.  Maybe I haven't thought about it enough, but it seems to me that a basic action predicate needs "return true" *for all* objects of the class that have moved.  In my mind the predicate needs to be evaluated many times, one for each possible object (or pair of objects, in the case of touches).  In other words, it seems to me that the loop over the objects has to occur *outside* eval(), not inside, so that we get a different boolean result for each object.